src/deployment/deploy.py

```diff
@@ -594,6 +602,47 @@ def assign_scaleset_identity_role(self) -> None:
             OnefuzzAppRole.ManagedNode,
         )
 
+    def assign_user_access(self) -> None:
+        logger.info("assinging user access to service principal")
+        app = get_application(
+            display_name=self.application_name,
+            subscription_id=self.get_subscription_id(),
+        )
+        user = get_signed_in_user(self.subscription_id)
+
+        if app:
+            sp = get_service_principal(app["appId"], self.subscription_id)
+            # Update appRoleAssignmentRequired if necessary
+            if not sp["appRoleAssignmentRequired"]:
```

Comment:
I don't think we should be changing the value of this when we create an assignment. 
The user might have intentionally change that value. 
Instead let's print a warning message indicating that this value is not set and unassigned users can still access the service

Reply:
Agreed and updated.